pkg/jsonhttp/jsonhttptest/jsonhttptest.go

```diff
@@ -11,12 +11,66 @@ import (
 	"io/ioutil"
 	"net/http"
 	"testing"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
 )
 
 func ResponseDirect(t *testing.T, client *http.Client, method, url string, body io.Reader, responseCode int, response interface{}) {
 	t.Helper()
 
-	resp := request(t, client, method, url, body, responseCode)
+	resp := request(t, client, method, url, body, responseCode, nil)
+	defer resp.Body.Close()
+
+	got, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got = bytes.TrimSpace(got)
+
+	want, err := json.Marshal(response)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !bytes.Equal(got, want) {
+		t.Errorf("got response %s, want %s", string(got), string(want))
+	}
+}
+
+// ResponseDirectWithJson checks for responses in json format. It is useful in cases where the response is json.
+func ResponseDirectWithJson(t *testing.T, client *http.Client, method, url string, body io.Reader, responseCode int, response interface{}) {
+	t.Helper()
+
+	resp := request(t, client, method, url, body, responseCode, nil)
+	defer resp.Body.Close()
+
+	got, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got = bytes.TrimSpace(got)
+
+	want, err := json.Marshal(response)
+	if err != nil {
+		t.Error(err)
+	}
+	var wantJson jsonhttp.StatusResponse
+	err = json.Unmarshal(want, &wantJson)
+	if err != nil {
+		t.Error(err)
+	}
+	wantString := "[" + wantJson.Message + "]"
+
+	if wantString != string(got) {
+		t.Errorf("got response %s, want %s", string(got), wantString)
+	}
+}
+
+func ResponseDirectWithHeaders(t *testing.T, client *http.Client, method, url string, body io.Reader, responseCode int,
+	response interface{}, headers map[string]string) {
```

Comment:
There is a specific convenient type `http.Headers` that should be used to pass the headers instead of map.

Reply:
Thanks. The http.Header is of `type Header map[string][]string`. But when i do a `req.Header.Get(string`)  it returns only string...` not []string`.  strange na..

Reply:
Yes, Header.Get returns the first value, but allows multiple values (properly encoded in the request), which can be accessed with Headers[headerName].